### PR TITLE
feat(media-container): bootstrap component - INNO-1153

### PR DESCRIPTION
**Pull request by degliwe**
# Specs
https://webgate.ec.europa.eu/CITnet/confluence/display/NEXTEUROPA/Media+container+Design+Specifications

### CODE REVIEW

**Comment by HAL-Patch-INNO** [comment]
Deploy preview for *europa-component-library* ready!

Built with commit 589d932aea225ba8b23ecf99f08a79c5afcf72d2

https://deploy-preview-823--europa-component-library.netlify.com

File: src/systems/ec/implementations/react/packages/media-container/MediaContainer.jsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+              key={track.src}
+              src={track.src}
+              kind={track.kind}
+              srcLang={track.srcLang}
+              label={track.label}
+            />
+          ))}
+        </video>
+      ) : (
+        <img className="ecl-media-container__media" src={image} alt="" />
```
Comment (emeryro) [inline src/systems/ec/implementations/react/packages/media-container/MediaContainer.jsx]: what if we want to have an alt text?
also we could put alt for video too

File: src/systems/ec/implementations/react/packages/media-container/MediaContainer.jsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  ),
+  className: PropTypes.string,
+};
+
+MediaContainer.defaultProps = {
+  description: '',
+  image: '',
+  className: '',
+  sources: '',
+  tracks: '',
```
Comment (emeryro) [inline src/systems/ec/implementations/react/packages/media-container/MediaContainer.jsx]: tracks and sources should be arrays

File: src/systems/ec/implementations/react/packages/media-container/package.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  },
+  "homepage": "https://github.com/ec-europa/europa-component-library",
+  "keywords": [
+    "ecl",
+    "europa-component-library",
+    "design-system",
+    "react"
+  ],
+  "dependencies": {
+    "@ecl/ec-react-component-button": "^2.0.0-alpha.0"
```
Comment (emeryro) [inline src/systems/ec/implementations/react/packages/media-container/package.json]: unused dependencies

File: src/systems/ec/implementations/react/packages/media-container/stories/Index.jsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+import MediaContainer from '../MediaContainer';
+
+storiesOf('MediaContainer', module)
+  .addDecorator(withKnobs)
+  .add('video', () => (
+    <MediaContainer
+      description={text('Description', demoContent.description)}
+      image={text('Image', demoContent.image)}
+      sources={demoContent.sources}
```
Comment (emeryro) [inline src/systems/ec/implementations/react/packages/media-container/stories/Index.jsx]: it is possible to have more complex knobs (see cards for instance), but this can be improved later

File: src/systems/ec/implementations/vanilla/packages/ec-component-media-container/ec-component-media-container.scss [hunk 5]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Media container component
+ * @define media
+ */
+
+// Import base and generic
+@import '@ecl/ec-base/ec-base';
+@import '@ecl/generic-component-media-container/generic-component-media-container';
```
Comment (emeryro) [inline src/systems/ec/implementations/vanilla/packages/ec-component-media-container/ec-component-media-container.scss]: we should not use generic components anymore.
scss is duplicated in ec and eu

File: src/systems/ec/implementations/vanilla/packages/ec-component-media-container/package.json [hunk 6]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "@ecl/ec-component-media-container",
+  "author": "European Commission",
+  "license": "EUPL-1.1",
+  "version": "2.0.0-alpha.0",
+  "description": "ECL Media container",
+  "style": "ec-component-media-container.scss",
+  "dependencies": {
+    "@ecl/ec-base": "^2.0.0-alpha.0",
+    "@ecl/ec-component-button": "^2.0.0-alpha.0",
```
Comment (emeryro) [inline src/systems/ec/implementations/vanilla/packages/ec-component-media-container/package.json]: unused dependencies

File: src/systems/ec/specs/components/media-container/demo/data--image.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+// Simple content for demo
```
Comment (emeryro) [inline src/systems/ec/specs/components/media-container/demo/data--image.js]: is this file needed?

File: src/systems/ec/specs/components/media-container/demo/data--image-shade.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+// Simple content for demo
```
Comment (emeryro) [inline src/systems/ec/specs/components/media-container/demo/data--image-shade.js]: is this file needed?

File: src/systems/ec/specs/components/media-container/docs/code.mdx [hunk 9]
```diff
@@ -0,0 +1,24 @@
+import Playground from '@ecl/website-components/Playground'
+import MediaContainer from '@ecl/ec-react-component-media-container/MediaContainer'
+
+import demoContent from '../demo/data';
+
+## Media container image
+
+<Playground playgroundLink="/storybook/ec/index.html?selectedKind=MediaContainer&selectedStory=image&stories=0">
```
Comment (emeryro) [inline src/systems/ec/specs/components/media-container/docs/code.mdx]: please use the new syntax for Playground (check cards for instance)

File: src/systems/generic/implementations/vanilla/generic-component-media-container/generic-component-media-container.scss [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+  .ecl-media-container__media {
+    display: block;
+    height: auto;
+    object-fit: cover;
+    width: 100%;
+  }
+
+  .ecl-media-container__caption {
+    font: $ecl-font-prolonged-m;
```
Comment (emeryro) [inline src/systems/generic/implementations/vanilla/generic-component-media-container/generic-component-media-container.scss]: I guess it is ok like this, but the specs makes no sense (they put prolonged in font-weight, and kept UI line height...)

**Review by emeryro — DISMISSED** [review]

File: src/systems/ec/implementations/vanilla/packages/ec-component-media-container/ec-component-media-container.scss [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+    background-color: $ecl-color-grey-10;
+    display: flex;
+    flex-direction: column;
+    margin: 0;
+  }
+
+  .ecl-media-container__media {
+    display: block;
+    height: auto;
+    object-fit: cover;
```
Comment (yhuard) [inline src/systems/ec/implementations/vanilla/packages/ec-component-media-container/ec-component-media-container.scss]: `object-fit` is not supported by IE and has partial support on Edge (`` only) https://caniuse.com/#search=object-fit

but the results look ok

File: src/systems/ec/specs/components/media-container/demo/data--video.js [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+module.exports = {
+  alt:
+    'The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.',
+  image:
+    'https://ec.europa.eu/education/sites/education/files/jean-monnet-gs-banner.jpg',
+  description:
+    'The European Commission has put forward ambitious yet realistic proposals for a modern EU budget. It is time for an EU budget that reflects rapid developments in innovation, the economy, the environment and geopolitics, amongst others.',
+  sources: [
+    {
+      src: 'http://clips.vorwaerts-gmbh.de/big_buck_bunny.mp4',
```
Comment (yhuard) [inline src/systems/ec/specs/components/media-container/demo/data--video.js]: can you use the `https` version instead? in order to avoid `Mixed Content: The page at 'https://deploy-preview-823--europa-component-library.netlify.com/storybook/ec/index.html?selectedKind=MediaContainer&selectedStory=video&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs' was loaded over HTTPS, but requested an insecure video 'http://clips.vorwaerts-gmbh.de/big_buck_bunny.webm'. This content should also be served over HTTPS.`

File: src/systems/ec/implementations/react/packages/media-container/MediaContainer.jsx [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+          controls="controls"
+          poster={image}
+          alt={alt}
+        >
+          {sources.map(source => (
+            <source key={source.src} src={source.src} type={source.type} />
+          ))}
+
+          {tracks.map(track => (
+            <track
```
[the remark this answers is not in the document]
Comment (yhuard) [reply src/systems/ec/implementations/react/packages/media-container/MediaContainer.jsx]: actually, React needs `srcLang` while the correct HTML attribute is `srclang`.

now, depending on your data, is you call your property `srcLang` and you can use `{...track}`, or you manually map `srcLang={track.srclang}`

either way, something needs to be done because right now, the property (in the data file) is called `srclang`, not `srcLang`

**Review by yhuard — CHANGES_REQUESTED** [review]

**Review by yhuard — APPROVED** [review]
